### ISSUE TITLE
Bug while debugging nodejs in ubuntu using snap installed version of node and code-insiders. Error is `The node version in "/snap/bin/node" is outdated(version), we require atleast Node 8.x`

### ISSUE THREAD

**Issue by yedhink** [microsoft/vscode#102355]
- VSCode Version: 1.48.0-insider
- OS Version: Linux x64 4.15.0-109-generic snap
- Node.js: 12.13.0
- Commit: e7920dce7bfd0b707ebfc0a5379c6edd2233e475

Steps to Reproduce:

1. Install [vscode-insiders from snap](https://snapcraft.io/code-insiders) in Ubuntu 18.04.4 LTS(bionic).
2. Install node using snap
3. create a folder test and a create a simple example.js file in it
4. open from terminal `code-insiders --disable-extensions test`
5. Start debugging - select either nodejs or nodejs(preview)
6. Error pops up saying `The node version in "/snap/bin/node" is outdated(version), we require atleast Node 8.x`
7. Tried `node --version` in integrated terminal - it shows `v12.18.1`
8. Tried `process.version` in dev tools console within vscode which gives `v12.13.0` ...its weird.
9. I dont use `nvm` to manage node versions. I just install from snap.

Does this issue occur when all extensions are disabled?: Yes

## Extra info that might help:

1. Within vscode using the dev tools i tried printing the `process.env.PATH`, which gave the following where both `/snap/bin` and `/snap/bin/node` were explicitly added by me.
```bash
"/home/ikigai/kde/kdesrc-build:/home/ikigai/.rbenv/shims:/home/ikigai/.rbenv/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/ikigai/.antigen/bundles/robbyrussell/oh-my-zsh/lib:/home/ikigai/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/git:/home/ikigai/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/npm:/home/ikigai/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/nvm:/home/ikigai/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/python:/home/ikigai/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/tmux:/home/ikigai/.antigen/bundles/romkatv/powerlevel10k:/home/ikigai/.local/bin:/home/ikigai/node_modules/.bin:/snap/bin:/usr/bin/yarn:/usr/local/go/bin:/snap/bin/node:/home/ikigai/.fzf/bin"
```
2. Attaching the `process.versions` output from dev tools within vscode:-
```json
node: "12.13.0"
v8: "8.0.426.27-electron.0"
uv: "1.32.0"
zlib: "1.2.11"
brotli: "1.0.7"
ares: "1.15.0"
modules: "76"
nghttp2: "1.41.0"
napi: "5"
llhttp: "1.1.4"
http_parser: "2.8.0"
openssl: "1.1.0"
icu: "65.1"
unicode: "12.1"
electron: "8.3.3"
microsoft-build: "5135798"
chrome: "80.0.3987.165"
```
3. Attaching a screenshot of the error
![Screenshot_20200713_112726](https://user-images.githubusercontent.com/31244999/87275566-01d8a780-c4fc-11ea-95a3-e11e26960560.png)

### Kindly let me know if anymore info is required from my side

**Comment by frastefanini** [comment]
Same here, with the latest vscode release (1.47), node 14.4.0 (installed from snap) and Pop!_OS (Ubuntu) 20.04.
Had to roll back to the legacy JS debugger.

**Comment by connor4312** [comment]
Thanks for the good issue! You can set `debug.javascript.usePreview: false` to use the old debugger until this is addressed.

**Comment by yedhink** [comment]
@connor4312  Thanks a lot. The above setting makes the debugger work. Out of curiosity, what exactly is the new in-preview javascript debugger? Would appreciate it if you have some links on which i could read more about this feature.

**Comment by connor4312** [comment]
@Tyriar suggested this is a snap bug, seems like https://github.com/nodesource/distributions/issues/843. Therefore I will assume that any node installed in /snap is 12 or higher (which is the large majority of users).

@yedhink check out https://code.visualstudio.com/updates/v1_47#_new-javascript-debugger

**Comment by mwoz123** [comment]
I'm having similar issue on node 14 with React app:
```
marcin@dell:~/workspace/barbershop$ node --version
v14.4.0
marcin@dell:~/workspace/barbershop$ whereis node
node: /snap/bin/node.npm /snap/bin/node.npx /snap/bin/node.yarn /snap/bin/node.yarnpkg /snap/bin/node
marcin@dell:~/workspace/barbershop$ uname -a
Linux dell 5.4.0-40-generic #44-Ubuntu SMP Tue Jun 23 00:01:04 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
marcin@dell:~/workspace/barbershop$ 
```
workaround with settings.json:
```
{
    "debug.javascript.usePreview": false
}
```

```
/snap/bin/node --inspect-brk=7731 node_modules/react-scripts/scripts/start.js --inspect-brk=39841 
ℹ ｢wds｣: Project is running at http://192.168.1.22/
PrefixFactory.js:55
ℹ ｢wds｣: webpack output is served from 
PrefixFactory.js:55
ℹ ｢wds｣: Content not from webpack is served from /home/marcin/workspace/barbershop/public
PrefixFactory.js:55
ℹ ｢wds｣: 404s will fallback to /
PrefixFactory.js:55
Starting the development server...

start.js:157
Error: EACCES: permission denied, read
Emitted 'error' event on ReadStream instance at:
    at internal/fs/streams.js:217:14
    at FSReqCallback.wrapper [as oncomplete] (fs.js:524:5)
```


`npm start` from vscode console works fine

@connor4312 can you help?

**Comment by connor4312** [comment]
@mwoz123 are you getting that `Error: EACCES: permission denied` both with and without `usePreview`?

**Comment by mwoz123** [comment]
My debug conf in `launch.json` 
```
        {
            "type": "node",
            "request": "launch",
            "name": "Debug ReactJS Dev server",
            "program": "${workspaceFolder}/node_modules/react-scripts/scripts/start.js",
            "skipFiles": [
                "<node_internals>/**"
            ],
            "args": ["--inspect-brk=39841"]
        },
```
It's unwrapped `"start": "react-scripts start",` which allowed previously (with node and vscode isntalled from deb) to debug in vscode react app

without the workaround in `settings.json` I'm getting  
```
The node version in "/snap/bin/node" is outdated(version), we require atleast Node 8.x
```
with workaround it went step further and fails at:
```
start.js:157
Error: EACCES: permission denied, read
```

https://github.com/facebook/create-react-app/blob/d2f813f8897ffcd2f0b0d2da75d0c44924c92f4d/packages/react-scripts/scripts/start.js#L157

**Comment by connor4312** [comment]
Okay, thanks for letting me know. This sounds like a (different) bug in Snap, or in Node's Snap. I'd recommend filing an issue on https://github.com/nodesource/distributions

**Comment by mwoz123** [comment]
@connor4312 as from commandline (even VSC termninal) :
- `node node_modules/react-scripts/scripts/start.js ` or  
- `npx react-scripts start` or 
- `npm start`

all of them start succefully react service, 

(but don't work when running from VSC debugger/`launch.json`)

I'm pretty sure it's VSCode related not Node.

**Comment by connor4312** [comment]
A fix for this issue has now been released in VS Code 1.47.2. If you previously had to opt-out of the new debugger, you should be safe to turn it back on. Thanks again for the report!

(Thanks for the bump mwoz, I will look at and address your issue presently)
